src/analysis/AnalysisClient.js

```diff
@@ -0,0 +1,113 @@
+const fs = require('fs');
+
+/* Class for handling product profitability analysis */
+class AnalysisClient {
+   
+  /*
+    Instance Attributes:
+      //estimated shipping per pound to amzn is 50 cents/pound) Found a 0.40 avg at 
+      //https://sellercentral.amazon.com/forums/t/how-much-does-it-cost-to-ship-fba-items-to-amazon/206820/7+
+      fbaShippingCostPerPound - current cost/pound to ship to amazon for FBA user;
+      approxTaxRate - accounts for tax on item when purchasing from walmart
+      effectiveValueOfOurDollar - because of sites like giftcardgranny.com and eBay, for every dollar we spend
+      at walmart, we could actually be spending less e.g. 98 cents.
+  */
+  constructor() {
+    this.fbaShippingCostPerPound = 0.50; 
+    this.approxTaxRate = 0.06;
+    this.analyzedProductsInfo = [];
+    this.effectiveValueOfDollar = 0.98;
+  }
+
+  //returns the total amount paid to walmart when item is purchased
+  getTotalAmountPaidToWalmart(walmartProduct) {
+    return (parseFloat(walmartProduct.price) + parseFloat(walmartProduct.shippingCost) + (this.approxTaxRate*parseFloat(walmartProduct.price)));
+  }
+  
+  //returns approx cost of shipping to AMZN as an FBA user which 
+  //will be paid whether we ship from home or from reseller
+  //in the future, this should also include approx cost of stocking items on amazon shelf
+  getApproxAmazonFBACost(amazonProduct) {
+    //using the ceiling function instead of round to avoid underestimation.
+    return (Math.ceil(parseFloat(amazonProduct.dimensions.weight['C$'])) * this.fbaShippingCostPerPound);
```

Comment:
What happens when the product weight is not given?

Reply:
In line 103 of the same file, products whose weights are not given are excluded.